internal/database/exiles.go

```diff
@@ -0,0 +1,132 @@
+package database
+
+import (
+	"context"
+	"time"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/naurffxiv/moddingway/internal/enum"
+)
+
+type AddExileEntryArgs struct {
+	DbUserID    int
+	Reason      string
+	ExileStatus enum.ExileStatus
+	StartTime 	string
+	EndTime 	string
+}
+
+func AddExileEntryTimed(conn *pgxpool.Pool, p AddExileEntryArgs) (int, error) {
+	query := `INSERT INTO exiles (userID, reason, exileStatus, startTimestamp, endTimestamp)
+	VALUES ($1, $2, $3, $4, $5)
+	RETURNING exileID`
+	
+	var exileID int
+	err := conn.QueryRow(
+		context.Background(),
+		query,
+		p.DbUserID,
+		p.Reason,
+		p.ExileStatus,
+		p.StartTime,
+		p.EndTime,
+	).Scan(&exileID)
+	
+	if err != nil {
+		return -1, err
+	}
+
+	return exileID, nil
+}
+
+func AddExileEntryIndefinite(conn *pgxpool.Pool, p AddExileEntryArgs) (int, error) {
+	query := `INSERT INTO exiles (userID, reason, exileStatus, startTimestamp)
+	VALUES ($1, $2, $3, $4)
+	RETURNING exileID`
+	
+	var exileID int
+	err := conn.QueryRow(
+		context.Background(),
+		query,
+		p.DbUserID,
+		p.Reason,
+		p.ExileStatus,
+		p.StartTime,
+	).Scan(&exileID)
+	
+	if err != nil {
+		return -1, err
+	}
+
+	return exileID, nil
+}
+
+type PendingUnexile struct {
+	ExileID	int
+	DbUserID string
+	ExileStatus enum.ExileStatus
+	DiscordUserID string
+	DiscordGuildID string
+}
+
+func GetPendingUnexiles(conn *pgxpool.Pool) ([]PendingUnexile, error) {
+	query := `SELECT e.exileID, e.userID, e.exileStatus, u.discordUserID, u.discordGuildID
+	FROM exiles e
+	JOIN users u ON e.userID = u.userID
+	WHERE e.exileStatus = $1 AND e.endTimestamp < $2;`
+
+	rows, err := conn.Query(context.Background(), query, enum.TimedExile, time.Now().UTC().Format(time.RFC3339))
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var pendingUnexiles []PendingUnexile
+	for rows.Next() {
+		currRow := PendingUnexile{}
+		err := rows.Scan(
+			&currRow.ExileID,
+			&currRow.DbUserID,
+			&currRow.ExileStatus,
+			&currRow.DiscordUserID,
+			&currRow.DiscordGuildID,
+		)
+		if err != nil {
+			return nil, err
+		}
+		pendingUnexiles = append(pendingUnexiles, currRow)
+	}
+
+	return pendingUnexiles, nil
+}
+
+func GetUserMostRecentExile(conn *pgxpool.Pool, dbUserID int) (int, error) {
+	query := `SELECT exileID FROM exiles
+	WHERE userID = $1
+	ORDER BY exileID DESC
+	LIMIT 1`
+
+	var exileID int
+	err := conn.QueryRow(context.Background(), query, dbUserID).Scan(&exileID)
+	if err != nil {
+		return -1, err
+	}
+	return exileID, nil
+}
```

Comment:
is there a case where we need this?

Reply:
It's used someone calls `/unexile` to remove the currently active exile
I could've used the first exile found, but I went with most recent in case someone gets exiled again after a failed exile where the exileStatus is set to `Unknown`.

Reply:
if it's unknown shouldn't we just drop the entry? otherwise that sounds good to me

Reply:
I think I'd want to keep it in for debugging purposes (and to figure out if anyone was somehow left dangling)

Reply:
if we want to use it for debugging purposes, we should move this to a test file (like `exiles-tests.go`) and make a unit test for it. if/when we need to debug this, we can just call this in the shell instead of actually incorporating our prod code with it

Reply:
There's a small misunderstanding, I meant keeping the DB entry for when something wrong happens as opposed to dropping the entry.

Reply:
im still kinda against that because we shouldnt be putting the database in a state where we have to use it to debug. that responsibility is on the code accessing the database, and not the database itself

Reply:
Alright, I guess I can just look at stdout if there are any issues, I'll go ahead and change this to GetUserExile()